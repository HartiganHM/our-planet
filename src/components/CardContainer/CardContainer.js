import React from 'react';
import './CardContainer.css';

const CardContainer = () => {
  return <div className="CardContainer">I am an CardContainer</div>;
};

export default CardContainer;
