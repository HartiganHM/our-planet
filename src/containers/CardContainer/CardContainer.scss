@import '../../styles/mixins.scss';
@import '../../styles/variables.scss';
.CardContainer {
  display: flex;
  flex-wrap: wrap;
  @include width(160);
  margin: auto;
  .content-placeholder {
    @include margin(1);
    @include contentDetailTitle();
    flex: 1;
    text-align: center;
  }
}

.filtered {
  @include width(160);
  margin: auto;
  .filtered-container {
    display: flex;
    flex-direction: column;
    .filter-header {
      @include contentHeader();
      @include margin(1);
      background-image: url('../../images/branding/Our-Planet-Light-Logo.png');
    }
    .wrapper {
      display: flex;
      flex-wrap: wrap;
      .content-placeholder {
        @include contentDetailTitle();
        flex: 1;
        text-align: center;
      }
    }
  }
}

@media screen and (max-width: 1634px) {
  .CardContainer {
    @include width(128);
  }
  .filtered {
    @include width(128);
  }
}

@media screen and (max-width: 1334px) {
  .CardContainer {
    @include width(96);
  }
  .filtered {
    @include width(96);
  }
}

@media screen and (max-width: 1014px) {
  .CardContainer {
    @include width(64);
  }
  .filtered {
    @include width(64);
  }
}

@media screen and (max-width: 694px) {
  .CardContainer {
    @include width(32);
  }
  .filtered {
    @include width(32);
  }
}
