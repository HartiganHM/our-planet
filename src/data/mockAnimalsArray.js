/*eslint-disable*/
export default [
  {
    name: 'Arctic Fox',
    status: 'Least Concern',
    population: 'Several Hundred Thousand',
    scientific_name: 'Vulpes lagopus',
    height: '11 inches',
    weight: '3 to 20 pounds',
    length: '18 to 27 inches',
    habitat: 'Antarctica / Arctic',
    facts:
      'The Arctic fox is primarily a carnivore that lives inland, away from the coasts. They are dependent on the presence of smaller animals (most often lemmings) to survive.  Arctic foxes also hunt for sea birds, fish, and other marine life. Smaller rodent populations waver between times of abundance and scarcity, which leaves the Arctic fox vulnerable when these creatures are low in numbers.',
    human_benefit:
      'The protection of the Arctic fox ensures the safety of a variety of other wildlife in the Arctic region.',
    image: 'arctic fox',
    display: true
  },
  {
    name: 'Arctic Wolf',
    status: 'Least Concern',
    population: '',
    scientific_name: 'Vulpes lagopus',
    height: '',
    weight: '70 to 175 pounds',
    length: '3.2 to 5.9 feet',
    habitat: 'Antarctica / Arctic',
    facts:
      'Often called the "polar wolf" or "white wolf," Arctic wolves inhabit the Arctic regions of North America and Greenland. Thanks to its isolation, the Arctic wolf is not threatened by hunting and habitat destruction in the same way as its southern relatives.',
    human_benefit:
      'Arctic wolves are carnivorous hunters. By nature they help to control the populations of other animals in the region like the musk ox, caribou and Arctic hares.',
    image: 'arctic wolf',
    display: true
  }
];
