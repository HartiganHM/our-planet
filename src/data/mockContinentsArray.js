/*eslint-disable*/
export default [
  {
    id: 1,
    name: 'Africa',
    bio:
      "Africa straddles the equator which bestows a rich diversity of wildlife and landscapes on the world's second largest continent. Large predators roam the sweeping savannas among huge herds of grazing herbivores, while apes, monkeys and snakes inhabit its dark, dense forests. Widely regarded as the place where human life originated, Africa is home to many of the world's best loved and most fascinating animals, as well as to some of its most endangered. Yet it's also home to many species that are as familiar to us in the UK as they are to Africans.",
    image_url:
      'https://our-planet.s3-us-west-2.amazonaws.com/continents/africa.jpg',
  },
  {
    id: 2,
    name: 'Antarctica / Arctic',
    bio: `Antarctica holds the record for being the coldest place on Earth, which together with extensive ice cover and lack of winter sunlight makes the continent pretty inhospitable to life. The sea, rather than the land, is what supports most creatures such as penguins and seals. It's often warmer than the land, so is a good place to retreat to and warm up to a balmy -2 Celsius. Permanently land-living plants and animals in Antarctica are small and ground-hugging to avoid wind chill and the accompanying dehydration.
  The Arctic ice cap first formed some three million years ago, providing a new habitat - polar pack ice - for mammals and birds to exploit. Seals and polar bears are now the primary inhabitants of this sea ice. The bears travel over it and the seals under, meeting at the seals\' breathing holes, to the detriment of the seals! On land, plants tough enough to survive the Arctic cold form the habitat known as tundra. Because of the harsh climate and prolonged darkness, many animals migrate away from the dark Arctic for the winter, returning to take advantage of the 24 hour summer sunshine to breed.`,
    image_url:
      'https://our-planet.s3-us-west-2.amazonaws.com/continents/antarctica+arctic.jpg',
  },
];
