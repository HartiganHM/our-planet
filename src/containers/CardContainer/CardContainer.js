import React, { Component } from 'react';
import './CardContainer.css';

class CardContainer extends Component {
  render() {
    return <div className="CardContainer">I am an CardContainer</div>;
  }
}

export default CardContainer;
