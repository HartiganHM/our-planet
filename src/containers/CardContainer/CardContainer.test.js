import React from 'react';
import { shallow } from 'enzyme';
import { CardContainer, mapStateToProps } from './CardContainer';
import mockAnimalsArray from '../../data/mockAnimalsArray';
import mockContinentsArray from '../../data/mockContinentsArray';

describe('CardContainer tests', () => {
  let renderedCardContainer;
  let animalsArray;
  let mockFilter;

  beforeEach(() => {
    mockFilter = 'default';
    animalsArray = mockAnimalsArray.map(animal => {
      return { ...animal, display: true };
    });
    renderedCardContainer = shallow(
      <CardContainer
        animals={animalsArray}
        continents={mockContinentsArray}
        filter={mockFilter}
      />
    );
  });

  it('Should match the snapshot', () => {
    expect(renderedCardContainer).toMatchSnapshot();
  });

  it('Should display Cards', () => {
    expect(renderedCardContainer.find('Card').length).toEqual(2);
  });

  it('Should not render certain cards if display is false', () => {
    animalsArray[0].display = false;
    renderedCardContainer = shallow(<CardContainer animals={animalsArray} />);

    expect(renderedCardContainer.find('Card').length).toEqual(1);
  });

  it('Should display different animals when receiving continentAnimals', () => {
    renderedCardContainer = shallow(
      <CardContainer
        animals={animalsArray}
        continentAnimals={[animalsArray[0]]}
      />
    );

    expect(renderedCardContainer.find('Card').length).toEqual(1);
  });

  it('Should display a placeholder if continentAnimals is empty', () => {
    const expectedElement = '.content-placeholder';
    const message = 'There are currently no endangered animals in this region';
    expect(renderedCardContainer.find(expectedElement).length).toEqual(0);

    renderedCardContainer = shallow(
      <CardContainer animals={animalsArray} continentAnimals={[]} />
    );
    expect(renderedCardContainer.find(expectedElement).length).toEqual(1);
    expect(renderedCardContainer.find(expectedElement).text()).toEqual(message);
  });

  it('Should display a placehold while the cards are loading', () => {
    const expectedElement = '.content-placeholder';
    const message = 'Loading';
    expect(renderedCardContainer.find(expectedElement).length).toEqual(0);

    renderedCardContainer = shallow(<CardContainer animals={[]} />);

    expect(renderedCardContainer.find(expectedElement).length).toEqual(1);
    expect(renderedCardContainer.find(expectedElement).text()).toEqual(message);
  });

  describe('mapStateToProps tests', () => {
    it('Should pull animals from the store', () => {
      const mockStore = {
        animals: mockAnimalsArray
      };
      const result = mapStateToProps(mockStore);

      expect(result.user).toEqual(mockStore.user);
    });
  });
});
