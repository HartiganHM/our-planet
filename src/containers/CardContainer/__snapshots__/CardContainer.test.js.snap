// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    animals={
        Array [
            Object {
              "continent_id": 2,
              "display": true,
              "facts": "The Arctic fox is primarily a carnivore that lives inland, away from the coasts. They are dependent on the presence of smaller animals (most often lemmings) to survive.  Arctic foxes also hunt for sea birds, fish, and other marine life. Smaller rodent populations waver between times of abundance and scarcity, which leaves the Arctic fox vulnerable when these creatures are low in numbers.",
              "habitat": "Antarctica / Arctic",
              "height": "11 inches",
              "human_benefit": "The protection of the Arctic fox ensures the safety of a variety of other wildlife in the Arctic region.",
              "length": "18 to 27 inches",
              "name": "Arctic Fox",
              "population": "Several Hundred Thousand",
              "scientific_name": "Vulpes lagopus",
              "status": "Least Concern",
              "weight": "3 to 20 pounds",
            },
            Object {
              "continent_id": 2,
              "display": true,
              "facts": "Often called the \\"polar wolf\\" or \\"white wolf,\\" Arctic wolves inhabit the Arctic regions of North America and Greenland. Thanks to its isolation, the Arctic wolf is not threatened by hunting and habitat destruction in the same way as its southern relatives.",
              "habitat": "Antarctica / Arctic",
              "height": "",
              "human_benefit": "Arctic wolves are carnivorous hunters. By nature they help to control the populations of other animals in the region like the musk ox, caribou and Arctic hares.",
              "length": "3.2 to 5.9 feet",
              "name": "Arctic Wolf",
              "population": "",
              "scientific_name": "Vulpes lagopus",
              "status": "Least Concern",
              "weight": "70 to 175 pounds",
            },
          ]
    }
    continents={
        Array [
            Object {
              "bio": "Africa straddles the equator which bestows a rich diversity of wildlife and landscapes on the world's second largest continent. Large predators roam the sweeping savannas among huge herds of grazing herbivores, while apes, monkeys and snakes inhabit its dark, dense forests. Widely regarded as the place where human life originated, Africa is home to many of the world's best loved and most fascinating animals, as well as to some of its most endangered. Yet it's also home to many species that are as familiar to us in the UK as they are to Africans.",
              "id": 1,
              "name": "Africa",
            },
            Object {
              "bio": "Antarctica holds the record for being the coldest place on Earth, which together with extensive ice cover and lack of winter sunlight makes the continent pretty inhospitable to life. The sea, rather than the land, is what supports most creatures such as penguins and seals. It's often warmer than the land, so is a good place to retreat to and warm up to a balmy -2 Celsius. Permanently land-living plants and animals in Antarctica are small and ground-hugging to avoid wind chill and the accompanying dehydration.
          The Arctic ice cap first formed some three million years ago, providing a new habitat - polar pack ice - for mammals and birds to exploit. Seals and polar bears are now the primary inhabitants of this sea ice. The bears travel over it and the seals under, meeting at the seals' breathing holes, to the detriment of the seals! On land, plants tough enough to survive the Arctic cold form the habitat known as tundra. Because of the harsh climate and prolonged darkness, many animals migrate away from the dark Arctic for the winter, returning to take advantage of the 24 hour summer sunshine to breed.",
              "id": 2,
              "name": "Antarctica / Arctic",
            },
          ]
    }
    filter="default"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          animalData={
                    Object {
                              "continent_id": 2,
                              "display": true,
                              "facts": "The Arctic fox is primarily a carnivore that lives inland, away from the coasts. They are dependent on the presence of smaller animals (most often lemmings) to survive.  Arctic foxes also hunt for sea birds, fish, and other marine life. Smaller rodent populations waver between times of abundance and scarcity, which leaves the Arctic fox vulnerable when these creatures are low in numbers.",
                              "habitat": "Antarctica / Arctic",
                              "height": "11 inches",
                              "human_benefit": "The protection of the Arctic fox ensures the safety of a variety of other wildlife in the Arctic region.",
                              "length": "18 to 27 inches",
                              "name": "Arctic Fox",
                              "population": "Several Hundred Thousand",
                              "scientific_name": "Vulpes lagopus",
                              "status": "Least Concern",
                              "weight": "3 to 20 pounds",
                            }
          }
/>,
        <Card
          animalData={
                    Object {
                              "continent_id": 2,
                              "display": true,
                              "facts": "Often called the \\"polar wolf\\" or \\"white wolf,\\" Arctic wolves inhabit the Arctic regions of North America and Greenland. Thanks to its isolation, the Arctic wolf is not threatened by hunting and habitat destruction in the same way as its southern relatives.",
                              "habitat": "Antarctica / Arctic",
                              "height": "",
                              "human_benefit": "Arctic wolves are carnivorous hunters. By nature they help to control the populations of other animals in the region like the musk ox, caribou and Arctic hares.",
                              "length": "3.2 to 5.9 feet",
                              "name": "Arctic Wolf",
                              "population": "",
                              "scientific_name": "Vulpes lagopus",
                              "status": "Least Concern",
                              "weight": "70 to 175 pounds",
                            }
          }
/>,
      ],
      "className": "CardContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "animalData": Object {
            "continent_id": 2,
            "display": true,
            "facts": "The Arctic fox is primarily a carnivore that lives inland, away from the coasts. They are dependent on the presence of smaller animals (most often lemmings) to survive.  Arctic foxes also hunt for sea birds, fish, and other marine life. Smaller rodent populations waver between times of abundance and scarcity, which leaves the Arctic fox vulnerable when these creatures are low in numbers.",
            "habitat": "Antarctica / Arctic",
            "height": "11 inches",
            "human_benefit": "The protection of the Arctic fox ensures the safety of a variety of other wildlife in the Arctic region.",
            "length": "18 to 27 inches",
            "name": "Arctic Fox",
            "population": "Several Hundred Thousand",
            "scientific_name": "Vulpes lagopus",
            "status": "Least Concern",
            "weight": "3 to 20 pounds",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "animalData": Object {
            "continent_id": 2,
            "display": true,
            "facts": "Often called the \\"polar wolf\\" or \\"white wolf,\\" Arctic wolves inhabit the Arctic regions of North America and Greenland. Thanks to its isolation, the Arctic wolf is not threatened by hunting and habitat destruction in the same way as its southern relatives.",
            "habitat": "Antarctica / Arctic",
            "height": "",
            "human_benefit": "Arctic wolves are carnivorous hunters. By nature they help to control the populations of other animals in the region like the musk ox, caribou and Arctic hares.",
            "length": "3.2 to 5.9 feet",
            "name": "Arctic Wolf",
            "population": "",
            "scientific_name": "Vulpes lagopus",
            "status": "Least Concern",
            "weight": "70 to 175 pounds",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            animalData={
                        Object {
                                    "continent_id": 2,
                                    "display": true,
                                    "facts": "The Arctic fox is primarily a carnivore that lives inland, away from the coasts. They are dependent on the presence of smaller animals (most often lemmings) to survive.  Arctic foxes also hunt for sea birds, fish, and other marine life. Smaller rodent populations waver between times of abundance and scarcity, which leaves the Arctic fox vulnerable when these creatures are low in numbers.",
                                    "habitat": "Antarctica / Arctic",
                                    "height": "11 inches",
                                    "human_benefit": "The protection of the Arctic fox ensures the safety of a variety of other wildlife in the Arctic region.",
                                    "length": "18 to 27 inches",
                                    "name": "Arctic Fox",
                                    "population": "Several Hundred Thousand",
                                    "scientific_name": "Vulpes lagopus",
                                    "status": "Least Concern",
                                    "weight": "3 to 20 pounds",
                                  }
            }
/>,
          <Card
            animalData={
                        Object {
                                    "continent_id": 2,
                                    "display": true,
                                    "facts": "Often called the \\"polar wolf\\" or \\"white wolf,\\" Arctic wolves inhabit the Arctic regions of North America and Greenland. Thanks to its isolation, the Arctic wolf is not threatened by hunting and habitat destruction in the same way as its southern relatives.",
                                    "habitat": "Antarctica / Arctic",
                                    "height": "",
                                    "human_benefit": "Arctic wolves are carnivorous hunters. By nature they help to control the populations of other animals in the region like the musk ox, caribou and Arctic hares.",
                                    "length": "3.2 to 5.9 feet",
                                    "name": "Arctic Wolf",
                                    "population": "",
                                    "scientific_name": "Vulpes lagopus",
                                    "status": "Least Concern",
                                    "weight": "70 to 175 pounds",
                                  }
            }
/>,
        ],
        "className": "CardContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "animalData": Object {
              "continent_id": 2,
              "display": true,
              "facts": "The Arctic fox is primarily a carnivore that lives inland, away from the coasts. They are dependent on the presence of smaller animals (most often lemmings) to survive.  Arctic foxes also hunt for sea birds, fish, and other marine life. Smaller rodent populations waver between times of abundance and scarcity, which leaves the Arctic fox vulnerable when these creatures are low in numbers.",
              "habitat": "Antarctica / Arctic",
              "height": "11 inches",
              "human_benefit": "The protection of the Arctic fox ensures the safety of a variety of other wildlife in the Arctic region.",
              "length": "18 to 27 inches",
              "name": "Arctic Fox",
              "population": "Several Hundred Thousand",
              "scientific_name": "Vulpes lagopus",
              "status": "Least Concern",
              "weight": "3 to 20 pounds",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "animalData": Object {
              "continent_id": 2,
              "display": true,
              "facts": "Often called the \\"polar wolf\\" or \\"white wolf,\\" Arctic wolves inhabit the Arctic regions of North America and Greenland. Thanks to its isolation, the Arctic wolf is not threatened by hunting and habitat destruction in the same way as its southern relatives.",
              "habitat": "Antarctica / Arctic",
              "height": "",
              "human_benefit": "Arctic wolves are carnivorous hunters. By nature they help to control the populations of other animals in the region like the musk ox, caribou and Arctic hares.",
              "length": "3.2 to 5.9 feet",
              "name": "Arctic Wolf",
              "population": "",
              "scientific_name": "Vulpes lagopus",
              "status": "Least Concern",
              "weight": "70 to 175 pounds",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
